### PR TITLE
Use subdir for URL

### LINKED ISSUE #15444: Migrate repository link in New Repository dialogue not using subdir from Base URL
- Gitea version (or commit ref): 1.15.0+dev-65-g0bb8bd819 built with GNU Make 4.3, go1.16.3 : bindata, timetzdata, sqlite, sqlite_unlock_notify
- Git version:
- Operating system: docker Linux (Debian)
  
  
  
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [x] No
- Log gist:

## Description

Setting the base url to a subdir e.g. `https://example.tld/git/` (tested in docker) then selecting create new repository it shows a link for migrations. This link points to the relative path /repo/migrate resolving to `https://example.tld/repo/migrate` instead of the correct `https://example.tld/git/repo/migrate`

## Screenshots

image
image

### PR DESCRIPTION
**Pull request by KN4CK3R**
Fixes #15444

### CODE REVIEW

**Review by lunny — APPROVED** [review]

**Review by silverwind — APPROVED** [review]

File: templates/user/auth/activate.tmpl [hunk 1]
```diff
@@ -19,7 +19,7 @@
 						{{end}}
 					{{else}}
 						{{if .NeedsPassword}}
-							<form class="ui form" action="/user/activate" method="post">
+							<form class="ui form" action="{{AppSubUrl}}/user/activate" method="post">
```
Comment (KN4CK3R) [inline templates/user/auth/activate.tmpl:22]: Here is a nested form. Thats not allowed by the HTML specs.

Comment (silverwind) [reply templates/user/auth/activate.tmpl:22]: Hmm can we just remove the inner form? They do `POST` do the same URL.

Comment (KN4CK3R) [reply templates/user/auth/activate.tmpl:22]: The inner form lacks the CSRF token and the outer form has no fields 🤷‍♂️

Comment (silverwind) [reply templates/user/auth/activate.tmpl:22]: @KN4CK3R should be block this PR because of this?

Comment (KN4CK3R) [reply templates/user/auth/activate.tmpl:22]: The inner form was added in #13631. I would keep the form for this PR. While testing if it causes troubles I found some more bugs I will address in a new PR.

Comment (zeripath) [reply templates/user/auth/activate.tmpl:22]: I think it's fine just to remove the inner form

Comment (KN4CK3R) [reply templates/user/auth/activate.tmpl:22]: #15473

**Comment by zeripath** [comment]
make lgtm work

**Pull request merged**